src/libraries/Microsoft.Extensions.DependencyInjection.Specification.Tests/src/ServiceProviderIsServiceSpecificationTests.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Extensions.DependencyInjection.Specification.Fakes;
+using Xunit;
+
+namespace Microsoft.Extensions.DependencyInjection.Specification
+{
+    public abstract class ServiceProviderIsServiceSpecificationTests
```

Comment:
Add `IServiceScopeProvider` and `IServiceProvider` test cases.

Reply:
> IServiceScopeProvider

IServiceScopeFactory

> IServiceProvider 

Not sure what you mean by this.

Reply:
```
            Assert.True(serviceProviderIsService.IsService(typeof(IServiceScopeProvider)));
            Assert.True(serviceProviderIsService.IsService(typeof(IServiceProvider)));
```

Reply:
Got it.

Reply:
And `Assert.True(serviceProviderIsService.IsService(typeof(IServiceProviderIsService)));`

Reply:
Done